pkg/unique-id/instance_registerer_test.go

```diff
@@ -0,0 +1,109 @@
+package unique_id_test
+
+import (
+	"context"
+	"errors"
+	"testing"
+	"time"
+
+	"github.com/applike/gosoline/pkg/clock"
+	"github.com/applike/gosoline/pkg/unique-id"
+	uniqueIdMocks "github.com/applike/gosoline/pkg/unique-id/mocks"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+type instanceRegistererSuite struct {
+	suite.Suite
+
+	clock      clock.FakeClock
+	repository *uniqueIdMocks.InstanceRepository
+	registerer unique_id.InstanceRegisterer
+}
+
+func TestRunInstanceRegistererSuite(t *testing.T) {
+	suite.Run(t, new(instanceRegistererSuite))
+}
+
+func (s *instanceRegistererSuite) SetupTest() {
+	date := time.Date(2021, 1, 2, 3, 4, 5, 6, time.UTC)
+
+	s.clock = clock.NewFakeClockAt(date)
+	s.repository = new(uniqueIdMocks.InstanceRepository)
+
+	settings := unique_id.Settings{
+		StartTime:    date,
+		MaxInstances: 3,
+	}
+
+	s.registerer = unique_id.NewInstanceRegisterWithInterfaces(s.clock, s.repository, settings)
+}
+
+func (s *instanceRegistererSuite) TeardownTest() {
+	s.repository.AssertExpectations(s.T())
+}
+
+// Test successful case
+func (s *instanceRegistererSuite) TestSuccess() {
+	ctx := context.Background()
+
+	instance := unique_id.Instance{
+		MachineId: 0,
+		CreatedAt: s.clock.Now(),
+	}
+
+	s.repository.
+		On("Persist", ctx, instance).
+		Return(nil)
+
+	registeredInstance, err := s.registerer.Register(ctx)
+	require.NoError(s.T(), err)
```

Comment:
What is the difference between `require.NoError`, `assert.NoError` and `s.NoError`? They all ensure there is no error, right?